internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -659,3 +664,52 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	xr.SetConditions(xpv1.Available())
 	return reconcile.Result{RequeueAfter: r.pollInterval}, errors.Wrap(r.client.Status().Update(ctx, xr), errUpdateStatus)
 }
+
+// EnqueueForCompositionRevisionFunc returns a function that enqueues (the
+// related) XRs when a new CompositionRevision is created. This speeds up
+// reconciliation of XRs on changes to the Composition by not having to wait for
+// the 60s sync period, but be instant.
+func EnqueueForCompositionRevisionFunc(of resource.CompositeKind, list func(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error, log logging.Logger) func(ctx context.Context, createEvent runtimeevent.CreateEvent, q workqueue.RateLimitingInterface) {
+	return func(ctx context.Context, createEvent runtimeevent.CreateEvent, q workqueue.RateLimitingInterface) {
+		rev, ok := createEvent.Object.(*v1.CompositionRevision)
+		if !ok {
+			// should not happen
+			return
+		}
+
+		// get all XRs
+		xrs := kunstructured.UnstructuredList{}
+		xrs.SetGroupVersionKind(schema.GroupVersionKind(of))
+		xrs.SetKind(schema.GroupVersionKind(of).Kind + "List")
+		if err := list(ctx, &xrs); err != nil {
+			// logging is most we can do here. This is a programming error if it happens.
+			log.Info("cannot list in CompositionRevision handler", "type", schema.GroupVersionKind(of).String(), "error", err)
+			return
+		}
+
+		// enqueue all those that reference the Composition of this revision
+		compName := rev.Labels[v1.LabelCompositionName]
+		if compName == "" {
+			return
+		}
+		for _, u := range xrs.Items {
+			xr := composite.Unstructured{Unstructured: u}
+
+			// only automatic
+			if pol := xr.GetCompositionUpdatePolicy(); pol != nil && *pol == xpv1.UpdateManual {
+				continue
+			}
+
+			// only those that reference the right Composition
+			compRef := meta.NamespacedNameOf(xr.GetCompositionReference())
+			if compRef.Name != compName {
```

Comment:
nit: I don't think it's needed to make it a NamespacedName
```suggestion
			if ref := xr.GetCompositionReference(); ref == nil || ref.Name != compName {
```